packages/repository/src/query.ts

```diff
@@ -173,14 +182,26 @@ export function isFilter(arg: any): arg is Filter {
  *   .build();
  * ```
  */
-export class WhereBuilder {
-  where: Where;
+export class WhereBuilder<MT = AnyObject> {
+  where: Where<MT>;
 
-  constructor(w?: Where) {
+  constructor(w?: Where<MT>) {
     this.where = w || {};
   }
 
-  private add(w: Where): this {
+  private add(w: Where<MT>): this {
```

Comment:
the logical conditions are handled by 'and' and 'or', so we may want to rename this method to something like 'addPredicate' or 'addComparison',